src/plugins/discover/public/plugin.tsx

```diff
@@ -233,6 +236,27 @@ export class DiscoverPlugin
       );
     }
 
+    if (plugins.globalSearch) {
+      const enableESQL = core.uiSettings.get(ENABLE_ESQL);
```

Comment:
Nit: I wonder if it's worth just adding this check to `if (plugins.globalSearch)` above and avoid registering it at all if ES|QL is disabled?

On the other hand, if we instead check `core.uiSettings.get(ENABLE_ESQL)` directly within the `find` method, then searching for ES|QL in global search would work as soon as it's enabled in advanced settings rather than needing a page refresh first (not sure if we actually care about this or not though).

Reply:
This is what I thought, is easier to test it also. Do you mind to leave it as it is?